mkosi/__init__.py

```diff
@@ -3661,6 +3662,16 @@ def make_generated_root(args: MkosiArgs, root: Path, for_cache: bool) -> Optiona
     return None
 
 
+def read_chunked(blob: BinaryIO) -> bytes:
+    '''Read a file or binary blob in chunks'''
+
+    while(True):
+        data = blob.read(DEFAULT_BUFFER_SIZE)
+        if not data:
+            break
+        yield data
```

Comment:
To somewhat shorten the loop (and make moving to the walrus operator later more mechanical)
```suggestion
    data = blob.read(DEFAULT_BUFFER_SIZE)
    while data
        yield data
        data = blob.read(DEFAULT_BUFFER_SIZE)
```

Reply:
I've migrated to `os.sendfile()` so this should now be redundant. If that doesn't work I'll look back to your suggested changes.